File: frontend-sistem-peminjaman-sederhana/lib/model/payment.dart
import 'package:apppeminjaman/model/PaymentId.dart';

class Payment {
  final PaymentId id;
  final double total;
  final int maxPaymentDate;
  final int? paymentDate;
  final String status;

  Payment({
    required this.id,
    required this.total,
    required this.maxPaymentDate,
    required this.paymentDate,
    required this.status,
  });

  factory Payment.fromJson(Map<String, dynamic> json) {
    return Payment(
      id: PaymentId.fromJson(json['id']),
      total: (json['total'] ?? 0).toDouble(),
      maxPaymentDate: json['maxPaymentDate'],
      paymentDate: json['paymentDate'],
      status: json['status'],
    );
  }
}

File: sistem-peminjaman-sederhana/src/main/java/com/SpringAPI/entities/Payment.java
package com.SpringAPI.entities;

import javax.persistence.*;
import java.util.Date;

@Entity
@Table(name = "payment")
public class Payment {

    @EmbeddedId
    private PaymentId id;
    
    @Column(name = "TOTAL")
    private double total;

    @Column(name = "MAXPAYMENTDATE")
    private Date maxPaymentDate;

    @Column(name = "PAYMENTDATE")
    private Date paymentDate;

    @Column(name = "STATUS")
    private String status;

	public PaymentId getId() {
		return id;
	}

	public void setId(PaymentId id) {
		this.id = id;
	}
	
	public double getTotal() {
		return total;
	}

	public void setTotal(double total) {
		this.total = total;
	}

	public Date getMaxPaymentDate() {
		return maxPaymentDate;
	}

	public void setMaxPaymentDate(Date maxPaymentDate) {
		this.maxPaymentDate = maxPaymentDate;
	}

	public Date getPaymentDate() {
		return paymentDate;
	}

	public void setPaymentDate(Date paymentDate) {
		this.paymentDate = paymentDate;
	}

	public String getStatus() {
		return status;
	}

	public void setStatus(String status) {
		this.status = status;
	}

	@Override
	public String toString() {
		return "Payment [id docno=" + id.getDocNo()+ " id index="+id.getId()+ ", total=" + total + ", maxPaymentDate=" + maxPaymentDate + ", paymentDate="
				+ paymentDate + ", status=" + status + "]";
	}
	
	

}
